dtfs-central-api/src/v1/controllers/utilisation-report-service/get-fee-records-to-key.controller/index.ts

```diff
@@ -0,0 +1,46 @@
+import { HttpStatusCode } from 'axios';
+import { Request, Response } from 'express';
+import { UtilisationReportRepo } from '../../../../repositories/utilisation-reports-repo';
+import { getBankNameById } from '../../../../repositories/banks-repo';
+import { ApiError, NotFoundError } from '../../../../errors';
+import { UtilisationReportWithFeeRecordsToKey } from '../../../../types/utilisation-reports';
+import { getFeeRecordsToKeyFromFeeRecordEntities } from './helpers';
+
+type GetFeeRecordsToKeyResponse = Response<UtilisationReportWithFeeRecordsToKey | string>;
+
+export const getFeeRecordsToKey = async (req: Request, res: GetFeeRecordsToKeyResponse) => {
+  const { reportId } = req.params;
+
+  try {
+    const utilisationReport = await UtilisationReportRepo.findOneByIdWithFeeRecordsFilteredByStatusWithPayments(Number(reportId), ['MATCH']);
+    if (!utilisationReport) {
+      throw new NotFoundError(`Failed to find a report with id ${reportId}`);
+    }
+
+    const { id, bankId, reportPeriod, feeRecords } = utilisationReport;
+
+    const bankName = await getBankNameById(bankId);
+    if (!bankName) {
+      throw new NotFoundError(`Failed to find a bank with id ${bankId}`);
+    }
+
+    const feeRecordsToKey = getFeeRecordsToKeyFromFeeRecordEntities(feeRecords);
+
+    return res.status(HttpStatusCode.Ok).send({
+      id,
+      bank: {
+        id: bankId,
+        name: bankName,
+      },
+      reportPeriod,
+      feeRecords: feeRecordsToKey,
+    });
```

Comment:
Is this return type getting typed checked against the expected response shape?

Reply:
Yes - the first argument of the `Response` generic is the `ResponseBody`